### PR TITLE
Upgrade TypeScript to 5.4

### PR DESCRIPTION
**Pull request by axelboc**
- https://devblogs.microsoft.com/typescript/announcing-typescript-5-2/
- https://devblogs.microsoft.com/typescript/announcing-typescript-5-3/
- https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,15 +1,12 @@
 {
   "compilerOptions": {
-    "module": "esnext",
-    "moduleResolution": "bundler",
-    "target": "esnext",
-    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "Preserve",
```
[the remark this answers is not in the document]
Comment (loichuder) [reply tsconfig.json:3]:
```suggestion
    "module": "preserve",
```

It looks weird when capitalized 🤨

**Review by loichuder — APPROVED** [review]

**Comment by loichuder** [comment]
Do you reckon the [NoInferType](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#support-for-require-calls-in---moduleresolution-bundler-and---module-preserve#the-noinfer-utility-type) could be useful somewhere ? I feel we run a few times into the problem it is trying to solve .

File: tsconfig.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (axelboc) [reply tsconfig.json]: Agreed, it's what the automcomplete suggests, but the docs are ambiguous. I'll keep the lower case, you're right.

**Comment by axelboc** [comment]
> Do you reckon the [NoInferType](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#support-for-require-calls-in---moduleresolution-bundler-and---module-preserve#the-noinfer-utility-type) could be useful somewhere ? I feel we run a few times into the problem it is trying to solve .

Definitely will be at some point, but it's tough to remember where it might have been in the past... Maybe in the mock utilities, with NX `axes` and `auxiliaries`, but don't think it's worth the time refactoring.

**Pull request merged**